src/composition.rs

```diff
@@ -580,13 +626,21 @@ impl Composition {
             flux::ast::walk::Node::File(&self.file),
         );
 
-        let mut analyzer = CompositionQueryAnalyzer {
-            bucket,
-            measurement,
-            fields: fields.unwrap_or_default(),
-            tag_values: tag_values.unwrap_or_default(),
+        // How to choose start position, in order of precedence:
+        //     1. position of previous composition
+        //     2. before all existing statements
+        //     3. if no stmts, then after imports are done
+        let mut start_position = match (
+            self.file.body.first(),
+            self.file.imports.last(),
+        ) {
+            (Some(stmt), _) => stmt.base().location.start,
+            (None, Some(import)) => ast::Position {
+                line: import.base.location.end.line + 1,
+                column: 0,
```

Comment:
Should it be `column: 1`? From my understanding, Monaco editor starts line and column from 1, unless you intentionally set it to be `0` here

Reply:
Ah, excellent catch!
Monaco-editor range is 1 indexed, but this is flux AST's `ast::Position`...which I believe is also 1 indexed! (Please confirm this Paul.)

Making the change.

Reply:
Good to know this is flux AST's `ast::Position`!

Reply:
[TextDocument positions are zero-based](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocuments).

Reply:
Excellent point.

So when we parse the AST for `from(bucket: "sample_data")`, the root ast::Position is 1 indexed.
```
"location": {
                "start": {
                    "line": 1,
                    "column": 1
                },
                "end": {
                    "line": 1,
                    "column": 28
                },
                "source": "from(bucket: \"sample_data\")"
            }
```

But since the applyEdit should be zero-based, shall I update the `lsp::WorkspaceEdit` to be zero-based?